src/System.Management.Automation/help/UpdatableHelpSystem.cs

```diff
@@ -1627,32 +1627,36 @@ internal static string LoadStringFromPath(PSCmdlet cmdlet, string path, PSCreden
         /// <returns></returns>
         internal static string GetFilePath(string path)
         {
+            FileInfo item = new FileInfo(path);
+
+            // We use 'FileInfo.Attributes' (not 'FileInfo.Exist')
+            // because we want to get exceptions
+            // like UnauthorizedAccessException or IOException.
+            if ((int)item.Attributes != -1)
+            {
+                return path;
+            }
 #if UNIX
             // On Linux, file paths are case sensitive.
             // The user does not have control over the files (HelpInfo.xml, .zip, and cab) that are generated by the Publishing team.
             // The logic below is to support updating help content via sourcepath parameter for case insensitive files.
-            FileInfo item = new FileInfo(path);
-            string directoryPath = item.Directory.FullName;
+            var dirInfo = item.Directory;
             string fileName = item.Name;
 
             // Prerequisite: The directory in the given path must exist and it is case sensitive.
-            if (Utils.DirectoryExists(directoryPath))
+            if (dirInfo.Exists)
             {
                 // Get the list of files in the directory.
-                string[] fileList = Directory.GetFiles(directoryPath);
-                foreach (string filePath in fileList)
+                FileInfo[] fileList = dirInfo.GetFiles(searchPattern: fileName, new System.IO.EnumerationOptions { MatchCasing = MatchCasing.CaseInsensitive });
+
+                // We use 'FileInfo.Attributes' (not 'FileInfo.Exist')
+                // because we want to get exceptions
+                // like UnauthorizedAccessException or IOException.
+                if (fileList.Length > 0 && (int)fileList[0].Attributes != -1)
```

Comment:
The original code doesn't check for existence for the files returned from `Direcotry.GetFiles`. Why do you add the check now? Can't we trust `direcotryInfo.GetFiles`?

Reply:
Please also respond to this comment: https://github.com/PowerShell/PowerShell/pull/7129#discussion_r204191483

Reply:
@iSazonov Can you please respond to the first comment in this thread (https://github.com/PowerShell/PowerShell/pull/7129#discussion_r204194315)?

Reply:
Sorry, GitHub sometimes tricky hides comments and I can not find where to type response :-)

You are right - the check is dead code - I removed it.

Reply:
Great, thanks! #Closed